### PR TITLE
Cluster setup scripts

### PR DESCRIPTION
**Pull request by skim0119**
## Purpose

- _Archive_ setup script (working version) for commonly-available clusters
- Enhance availability for other collaborators

## Feature

- `support/cluster` directory to add cluster setup scripts.
  - add `expanse` setup script
- `readme` file.

## Description

- dependencies (HDF5-Parallel, NeuroH5, Neuron) built with a common compiler
- paths setup as a module

```bash
$ module avail
------------------------------------------------- test/modules -------------------------------------------------
   miv-simulator (L)
```
```bash
$ module help miv-simulator

----------------------------------- Module Specific Help for "miv-simulator" -----------------------------------
This module is expected to be used with python-environment conda_env/miv.
```
```bash
$ module whatis miv-simulator
miv-simulator       : Require C/C++ libraries and Python modules for MiV-Simulator.
miv-simulator       : The module includes phdf5-1.12.1, NEURON-8.2.1, and NeuroH5
miv-simulator       : Loading this module will additionally load openmpi, cmake, and anaconda.
```

## TODO

- [x] Check the installation in compute node
- [x] Add link in documentation

### CODE REVIEW

File: docs/tutorial/installation.md [hunk 1]
```diff
@@ -146,3 +148,8 @@ cmake .
 make -j4
 export PATH=<NeuroH5 installation path>/bin:$PATH
 ```
+
+## Tools and scripts to aid cluster usage
+
+- [How to install and configure as module](https://github.com/GazzolaLab/MiV-Simulator/tree/latest/support/cluster)
+- [Machinable: A modular system to manage research code](https://machinable.org://machinable.org/)
```
Comment (frthjf) [inline docs/tutorial/installation.md:155]: URL typo?

**Review by frthjf — APPROVED** [review]
Neat! LGTM.

One thing, this does not get packaged right, so it's just part of the repo for people to copy/adapt? If so, I wonder if it should just be part of the docs for people to copy & paste?

File: docs/tutorial/installation.md (hunk 1 above)
Comment (skim0119) [reply docs/tutorial/installation.md]: 8550688b989e744108c85a7114b0b47db8811d86

**Comment by skim0119** [comment]
> One thing, this does not get packaged right, so it's just part of the repo for people to copy/adapt? If so, I wonder if it should just be part of the docs for people to copy & paste?

Ideally, it should still work on the targetted cluster, which is why I kept it in the main directory. I agree that it can be part of the documentation for people to copy & paste, but I wanted the script file to be more visible on github repo.

**Review by iraikov — APPROVED** [review]
I suggest enabling zlib compression support in the HDF5 installation (the relevant cmake options are -DHDF5_ENABLE_PLUGIN_SUPPORT=ON -DHDF5_ENABLE_Z_LIB_SUPPORT=ON). I usually use compression with larger files. Also, does the miv_simulator module file include any dependencies such as MPI that need to be loaded? Other than that, this looks good to me.

**Comment by skim0119** [comment]
> I suggest enabling zlib compression support in the HDF5 installation (the relevant cmake options are -DHDF5_ENABLE_PLUGIN_SUPPORT=ON -DHDF5_ENABLE_Z_LIB_SUPPORT=ON). I usually use compression with larger files. Also, does the miv_simulator module file include any dependencies such as MPI that need to be loaded? Other than that, this looks good to me.

Okay, I'll add the suggestions. Thanks!

**Pull request merged**